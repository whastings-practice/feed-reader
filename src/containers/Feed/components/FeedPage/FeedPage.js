import React from 'react';

import ListGroup from 'reactstrap/lib/ListGroup';
import ListGroupItem from 'reactstrap/lib/ListGroupItem';

import Post from '../Post/Post';

export default function FeedPage(props) {
  const { feed } = props;

  return (
    <section>
      <h2>{feed.title}</h2>
      <ListGroup>
        {feed.posts.map((post) => (
          <ListGroupItem key={post.link}>
            <Post post={post} />
          </ListGroupItem>
        ))}
      </ListGroup>
    </section>
  );
}
