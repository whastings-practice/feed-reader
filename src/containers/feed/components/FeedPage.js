import React from 'react';

export default function FeedPage(props) {
  const { feed } = props;

  if (!feed.posts) {
    return null;
  }

  return (
    <section>
      <h2>{feed.title}</h2>
      <ul>
        {feed.posts.map((post) => (
          <li key={post.url}>
            <a href={post.url}>{post.title}</a>
          </li>
        ))}
      </ul>
    </section>
  );
}
