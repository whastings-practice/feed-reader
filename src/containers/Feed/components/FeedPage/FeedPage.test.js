import React from 'react';
import { shallow } from 'enzyme';

import FeedPage from './FeedPage';

describe('FeedPage', () => {
  let props;
  let wrapper;

  function render() {
    wrapper = shallow(<FeedPage {...props} />);
  }

  beforeEach(() => {
    props = {
      feed: {
        title: 'Foo Bar',
        posts: [
          { title: 'Foo', link: '/foo' },
          { title: 'Bar', link: '/bar' },
        ],
      },
    };
    render();
  });

  test('it renders a page with a list of posts', () => {
    expect(wrapper).toMatchSnapshot();
    const posts = wrapper.find('Post');
    props.feed.posts.forEach((post, i) => {
      expect(posts.at(i)).toHaveProp('post', post);
    });
  });
});
