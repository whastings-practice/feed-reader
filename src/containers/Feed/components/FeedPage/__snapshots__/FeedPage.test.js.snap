// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedPage it renders a page with a list of posts 1`] = `
<section>
  <h2>
    Foo Bar
  </h2>
  <ListGroup
    tag="ul"
  >
    <ListGroupItem
      key="/foo"
      tag="li"
    >
      <Post
        post={
          Object {
            "link": "/foo",
            "title": "Foo",
          }
        }
      />
    </ListGroupItem>
    <ListGroupItem
      key="/bar"
      tag="li"
    >
      <Post
        post={
          Object {
            "link": "/bar",
            "title": "Bar",
          }
        }
      />
    </ListGroupItem>
  </ListGroup>
</section>
`;
